lib/rucio/daemons/storage/consistency/actions.py

```diff
@@ -0,0 +1,805 @@
+# -*- coding: utf-8 -*-
+# Copyright 2013-2020 CERN
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Authors:
+# - Stefan Piperov <stefan.piperov@cern.ch>, 2020-2021
+# - Igor Mandrichenko <ivm@fnal.gov>, 2020-2021
+
+"""
+Storage-Consistency-Actions is a daemon to delete dark files, and re-subscribe the missing ones, identified previously in a Storage-Consistency-Scanner run.
+"""
+
+import traceback
+
+import logging
+import os
+import socket
+import threading
+from copy import deepcopy
+from datetime import datetime, timedelta
+from random import randint
+from re import match
+import sys 
+import glob
+import json
+import time
+import gzip
+import re
+import csv
+
+
+from rucio.core.rse import list_rses, get_rse_id
+from rucio.rse.rsemanager import lfns2pfns, get_rse_info, parse_pfns
+from rucio.common.config import config_get
+
+from rucio.common.types import InternalAccount, InternalScope
+from rucio.core.replica import __exists_replicas, update_replicas_states
+from rucio.core.quarantined_replica import add_quarantined_replicas
+from rucio.core.monitor import record_gauge, record_counter, record_timer, MultiCounter
+
+
+from rucio.common.utils import daemon_sleep
+from rucio.common import exception
+from rucio.common.logging import formatted_logger, setup_logging
+from rucio.common.exception import DatabaseException, UnsupportedOperation, RuleNotFound
+from rucio.core.heartbeat import live, die, sanity_check
+from rucio.db.sqla.util import get_db_time
+
+
+
+##########################################################################
+### NOTE: these are needed by local version of declare_bad_file_replicas()
+### remove after move to core/replica.py
+from rucio.db.sqla.session import transactional_session
+from rucio.db.sqla.constants import (ReplicaState, BadFilesStatus)
+##########################################################################
+
+graceful_stop = threading.Event()
+
+
+##########################################################################
+### NOTE: declare_bad_file_replicas will be used directly from core/replica.py
+### when handling of DID is added there
+@transactional_session
+def declare_bad_file_replicas(dids, rse_id, reason, issuer,
+     status=BadFilesStatus.BAD, scheme='srm', session=None):
```

Comment:
srm default? shouldn't be something different?

Reply:
scheme is completely unused. It can be removed entirely if we're planning on replacing it with the https://github.com/rucio/rucio/pull/5068 version. I assume it's their to retain API compatibility with the old method?

So either remove it or just default it to None so people stop looking. :-)

Reply:
Replaced with 'None'